adobe_vipm/flows/utils.py

```diff
@@ -928,3 +929,30 @@ def exclude_subscriptions_with_deployment_id(adobe_subscriptions):
     ]
     adobe_subscriptions["items"] = items
     return adobe_subscriptions
+
+
+def is_deployment_id_gc_main_agreement_deployment(
+    gc_main_agreement,
+    deployment_id,
+    product_id,
+):
+    gc_transfer_id = ""
+    gc_deployment_id = ""
+    if gc_main_agreement and deployment_id:
+        gc_main_agreement_transfer_id = gc_main_agreement.get("transfer_id", "")
+        gc_main_agreement_id = gc_main_agreement.get("main_agreement_id", "")
+        gc_deployments = get_gc_agreement_deployments_by_main_agreement(
+            product_id, gc_main_agreement_id
+        )
+        if gc_deployments:
+            gc_deployment = next(
+                x for x in gc_deployments if x["deployment_id"] == deployment_id
+            )
+            gc_transfer_id = gc_deployment["transfer_id"] if gc_deployment else ""
```

Comment:
Honestly speaking by defining `gc_deployment` as an default empty dict `{}` these two lines can be simplified to

```
gc_transfer_id = gc_deployment.get("transfer_id", "")
```